link/cgroup_test.go

```diff
@@ -75,3 +84,66 @@ func TestLinkCgroup(t *testing.T) {
 		loadPinned: LoadPinnedCgroup,
 	})
 }
+
+func TestCgroupPerCPUStorageMarshaling(t *testing.T) {
+	numCPU, err := internal.PossibleCPUs()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if numCPU < 2 {
+		t.Skip("Test requires at least two CPUs")
+	}
+	testutils.SkipOnOldKernel(t, "4.20", "per-CPU CGoup storage")
+
+	arr, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type:      ebpf.PerCPUCGroupStorage,
+		KeySize:   16,
+		ValueSize: 8,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer arr.Close()
+
+	prog := mustCgroupEgressProgramWithMap(t, arr)
+	cgroup := mustCgroupFixturesWithProgram(t, prog)
+
+	link, err := AttachCgroup(CgroupOptions{
+		Path:    cgroup.Name(),
+		Attach:  ebpf.AttachCGroupInetEgress,
+		Program: prog,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer link.Close()
+
+	cgroupStat := unix.Stat_t{}
+	err = unix.Fstat(int(cgroup.Fd()), &cgroupStat)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var mapKey = &BpfCgroupStorageKey{
+		CgroupInodeId: cgroupStat.Ino,
+		AttachType:    ebpf.AttachCGroupInetEgress,
+	}
+
+	values := []uint64{1, 2}
```

Comment:
This needs to have length numCPU, I think. Otherwise the test will fail on machines with more than two CPUs.
```suggestion
	values := make([]uint64, numCPU)
	copy(values, []uint64{1, 2})
```

Reply:
This does not appear to be necessary. This test passes fine on my machine with 4 cpus.